tests/unit/test_table.py

```diff
@@ -102,6 +102,21 @@ def test_ctor_with_key(self):
         self.assertEqual(encryption_config.kms_key_name, self.KMS_KEY_NAME)
 
 
+class TestTableBase:
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.bigquery.table import TableBase
+
+        return TableBase
+
+    def _make_one(self, *args, **kw):
+        return self._get_target_class()(*args, **kw)
+
+    def test_ctor_defaults(self):
+        instance = self._make_one()
+        assert instance._properties == {}
+
+
```

Comment:
ISTM we need direct tests of the shared properties (`project`, `dataset_id`, `table_id`, `path`) and dunder-methods (`__eq__`, `__hash__`).

Reply:
These are already tested through the individual Table* classes' tests, the common logic has just been hoisted into a base class. Or do you feel there's still value in additional tests specifically targeting `TableBase`?

Reply:
I'd rather have explicit tests tied to the implementatoin class.  If the derived classes no longer have the methods, then they shouldn't need to test them.

Reply:
I'm ambivalent about this. In some ways it's nice just to test the "public" interface (subclasses). That way if we refactor again, code coverage can let us know when there's dead code in the base class.

That said, maybe we want `TableBase` to be public to simplify some of our `Union` types in the client?

Reply:
I'd say the dilemma stems from the fact that the base class was created after the now-subclasses, resulting in subclasses having dedicated tests, and no explicit base class tests.

If it was the other way around and the base class would be added at the same time as the other `Table*` classes, I'm pretty sure we would not be writing several copies of tests (one per subclass) for the common properties.

Taking the same point to the extreme, all classes inherit from `object`, but we don't actually test things such as their default `__str_()` or `__hash__()` behavior, unless our class explicitly overrides the default implementation.

I thus lean more towards refactoring the tests as well.